packages/redis/src/RedisClient.ts

```diff
@@ -153,4 +222,319 @@ export class RedisClient implements IRequireInitialization, IDisposable {
             span.finish();
         }
     }
+
+    public async xAddObject<T>(
+        context: SpanContext,
+        type: string | IClassType<T>,
+        streamName: string,
+        key: string,
+        body: T,
+        id: string = AutoGenerateRedisStreamID
+    ): Promise<string> {
+        const db = this.config.db;
+        const span = this.tracer!.startSpan("Redis Client xAddObject Call", { childOf: context });
+
+        this.spanLogAndSetTags(span, this.xAddObject.name, db, key, streamName);
+
+        const typeName = this.getTypeName(type);
+
+        const encodedBody = this.encoder.encode({
+            type: typeName,
+            payload: body,
+        });
+
+        const buf = Buffer.from(encodedBody);
+        const storableValue = this.config.base64Encode ? buf.toString("base64") : buf;
+        try {
+            const insertedId = await this.client.xadd(
+                streamName,
+                id,
+                key,
+                storableValue,
+                "type",
+                typeName
+            );
+            this.metrics!.increment(RedisMetrics.XAdd, {
+                type,
+                db,
+                streamName,
+                result: RedisMetricResults.Success,
+            });
+
+            return insertedId;
+        } catch (e) {
+            failSpan(span, e);
+            this.metrics!.increment(RedisMetrics.XAdd, {
+                type,
+                db,
+                streamName,
+                result: RedisMetricResults.Error,
+                error: e,
+            });
+            throw e;
+        } finally {
+            span.finish();
+        }
+    }
+
+    public async xGroup(
+        context: SpanContext,
+        streamName: string,
+        consumerGroup: string,
+        consumerGroupStartId: string,
+        supressAlreadyExistsError: boolean = true
+    ): Promise<string> {
+        const db = this.config.db;
+        const span = this.tracer.startSpan("Redis Client xGroup Call", { childOf: context });
+        this.spanLogAndSetTags(span, this.xGroup.name, this.config.db, undefined, streamName);
+        try {
+            const response = await this.client.xgroup([
+                "create",
+                streamName,
+                consumerGroup,
+                consumerGroupStartId,
+                "mkstream",
+            ]);
+            this.metrics.increment(RedisMetrics.XGroupCreate, {
+                db,
+                streamName,
+                consumerGroup,
+                result: RedisMetricResults.Success,
+            });
+            return response;
+        } catch (err) {
+            const alreadyExistsErrorMessage = "BUSYGROUP Consumer Group name already exists";
+            if (supressAlreadyExistsError && err.message === alreadyExistsErrorMessage) {
+                // TODO should this be considered an error or success?
+                this.metrics.increment(RedisMetrics.XGroupAlreadyExists, {
```

Comment:
regarding the TODO ... do we even need a separate metric for this? can't think of any use case where somebody would be interested in this number ... cc @chrnola can you think of a use case?

Reply:
I think it's worth tracking, but not with its own time series. I would vote for incrementing the `RedisMetrics.XGroupCreate` counter here but swapping the `reason` label for a new `RedisMetricResults.AlreadyExists` instead, something like:
```suggestion
                this.metrics.increment(RedisMetrics.XGroupCreate, {
                    ...
                    result: RedisMetricResults.AlreadyExists,
                });
```